### DISCUSSION TITLE
Healthchecks.oi hooks on Windows

### DISCUSSION THREAD

**Discussion by lerufino** [garethgeorge/backrest#532] [Q&A]
Hi!
I'm unsure if this is the right place to ask for help with this, but I am out of good options now.
Im doing some tests and learning about Backrest and Healthchecks.io, I'm new to both. I tried 2 ways, neither worked:

1. **Direc integration with Healthchecks.io, like this:**

`{
   "conditions":  ["CONDITION_SNAPSHOT_END"],
   "actionHealthchecks":  {
   "webhookUrl":  "https://hc-ping.com/API_KEY",
   "template":  "{{ .Summary }}"
  }
}`

Hook Output:
`[tasklog] 10:29:01.622Z	info	run hook healthchecks hook plan/plan1/hook/1	Sending healthchecks message to https://hc-ping.com/API_KEY
[tasklog] 10:29:01.623Z	debug	run hook healthchecks hook plan/plan1/hook/1	Sending healthchecks message	{"payload": "\nBackrest Snapshot Notification\nTask: backup for plan \"plan1\" at 2024-10-25T10:29:01-03:00\nEvent: snapshot end\nSnapshot: 56789...3501c\nOverview:\n- Data added: 0.000 B\n- Total files processed: 3\n- Total bytes processed: 0.000 B\n\nBackup Statistics:\n- Files new: 0\n- Files changed: 0\n- Files unmodified: 3\n- Dirs new: 0\n- Dirs changed: 0\n- Dirs unmodified: 3\n- Data blobs: 0\n- Tree blobs: 0\n- Total duration: 0.0449748s\n\n"}
[tasklog] 10:29:02.313Z	debug	run hook healthchecks hook plan/plan1/hook/1	Healthchecks response	{"body": "OK"}
[tasklog] 10:29:02.313Z	info	run hook healthchecks hook plan/plan1/hook/1	task finished	{"duration": 0.6904047}`

2. **Command through CMD, like this:**

`{
  "conditions":  ["CONDITION_SNAPSHOT_END"],
  "actionCommand":  {
    "command":  "#! cmd.exe\nC:\\Program Files\\Backrest\\commands\\HealthcheksTest.bat"
  }
}`

Hook Output:
`[script] [cmd.exe]
[script] C:\Program Files\Backrest\commands\HealthcheksTest.bat
[output] Microsoft Windows [vers�o 10.0.19045.5011]
[output] (c) Microsoft Corporation. Todos os direitos reservados.
[output] 
[output] C:\Program Files\Backrest>Mais? [tasklog] 10:29:01.609Z	info	run hook command hook plan/plan1/hook/0	task finished	{"duration": 0.0206051}`

With contente like: `curl -X POST -d "temperature=-7" https://hc-ping.com/API_KEY`

-------------

It does not log erros on execution, does not 'start' a new Check.
If I start the Check on browser or just executing the .bat i starts logging on Healthchecks.io but does not consider the ping as alive. It looks like this:
![image](https://github.com/user-attachments/assets/2944cb49-7726-4f19-a170-57263af42f03)

The Ok on the image is from when I started ir manually.

I believe I am missing something, but for the life of me can't figure it out by myself!

**Comment by lerufino** [comment]
I'd like to add that I made a similar question about the same thing in Healthcheck.io's Github Discussion, if anyone want to take a look from their point of view.

[Link to Discussion](https://github.com/healthchecks/healthchecks/discussions/1079#discussioncomment-11064680)

**Comment by Chucky2401** [comment]
I just implemented Healthchecks 2 days ago, and everything works great on my side.

![image](https://github.com/user-attachments/assets/32e97e2f-79a9-4a6c-99f5-482fa05bbf49)

But, I have different conditions on Backrest side.
Here what I have:

```json
"hooks": [
    {
      "conditions": [
        "CONDITION_SNAPSHOT_START",
        "CONDITION_SNAPSHOT_ERROR",
        "CONDITION_SNAPSHOT_WARNING",
        "CONDITION_SNAPSHOT_SUCCESS"
      ],
      "actionHealthchecks": {
        "webhookUrl": "https://health.domain.local/ping/00x00x00-00xx-xx00-00xx-00x00x00x00x",
        "template": "{{ .Summary }}"
      }
    }
```

Try to maybe just add the condition `CONDITION_SNAPSHOT_START` as your webhookurl seems good in your output.
